mongodb/src/main/java/org/hibernate/ogm/datastore/mongodb/query/parsing/nativequery/impl/NativeQueryParser.java

```diff
@@ -92,12 +94,13 @@ public Rule Separator() {
 	}
 
 	public Rule Reserved() {
-		return FirstOf( Find(), FindOne(), FindAndModify(), Insert(), Remove(), Update(), Count() );
-		//TODO There are many more query types than what we support.
+		return FirstOf( Aggregate(), Find(), FindOne(), FindAndModify(), Insert(), Remove(), Update(), Count() );
```

Comment:
Let's add it at the end instead of at the beginning. It's less common than the other operations.